ams/ResetCommand.cs

```diff
@@ -32,67 +32,87 @@ internal class ResetCommand : BaseMigrator
 
         public override async Task MigrateAsync(CancellationToken cancellationToken)
         {
-            MediaServicesAccountResource? account = null;
-            try
+            var (isAMSAcc, account) = await IsAMSAccountAsync(_options.AccountName, cancellationToken);
+            if (!isAMSAcc)
             {
-                account = await GetMediaAccountAsync(_options.AccountName, cancellationToken);
+                var (storageClient, accountId) = await _resourceProvider.GetStorageAccount(_options.AccountName, cancellationToken);
+
+                _logger.LogInformation("Begin reset storage container: {name}", storageClient.AccountName);
+                double totalItems = await GetStorageBlobMetricAsync(accountId, cancellationToken);
+                var containers = storageClient.GetBlobContainersAsync(
+                               cancellationToken: cancellationToken);
+                _logger.LogInformation("The total containers count of the storage account is {count}.", totalItems);
+                List<BlobContainerItem>? filteredList = await containers.ToListAsync();
+                int resetAssetCount = 0;
+                foreach (var container in filteredList)
+                {
+                    BlobContainerClient containerClient = storageClient.GetBlobContainerClient(container.Name);
+                    await MigrateTaskAsync(containerClient, resetAssetCount, cancellationToken);
+                }
             }
-            catch (Exception)
+            else if (account != null)
+            {
+                _logger.LogInformation("Begin reset assets on account: {name}", account.Data.Name);
+                await _resourceProvider.SetStorageResourceGroupsAsync(account, cancellationToken);
+                AsyncPageable<MediaAssetResource> assets = account.GetMediaAssets()
+                    .GetAllAsync(cancellationToken: cancellationToken);
+                List<MediaAssetResource>? assetList = await assets.ToListAsync(cancellationToken);
+                int resetAssetCount = 0;
+                foreach (var asset in assetList)
+                {
+                    var (storage, _) = await _resourceProvider.GetStorageAccount(asset.Data.StorageAccountName, cancellationToken);
+                    var container = storage.GetContainer(asset);
+                    if (!await container.ExistsAsync(cancellationToken))
+                    {
+                        _logger.LogWarning("Container {name} missing for asset {asset}", container.Name, asset.Data.Name);
+                        return;
+                    }
+
+                    await MigrateTaskAsync(container, resetAssetCount, cancellationToken);
+                }
+                _logger.LogDebug("{resetAssetCount} out of {totalAssetCount} assets has been reset.", resetAssetCount, assetList.Count);
+            }
+            else
```

Comment:
The code here should be moved to the place where it is not an AMS account, nor Storage Account.  A good place is to move to after line 39, when it is not a storage account, throw the exception.